testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -265,8 +264,19 @@ impl BlockMinerThread {
         let stackerdb_contracts = stackerdbs
             .get_stackerdb_contract_ids()
             .expect("FATAL: could not get the stacker DB contract ids");
-        // TODO: get this directly instead of this jankiness when .signers is a boot contract
-        let signers_contract_id = boot_code_id(SIGNERS_NAME, self.config.is_mainnet());
+
+        let reward_cycle = self
+            .burnchain
+            .block_height_to_reward_cycle(self.burn_block.block_height)
+            .expect("FATAL: no reward cycle for burn block");
+
+        let signers_contract_id = boot_code_id(
+            &make_signers_db_name(
+                u32::try_from(reward_cycle % 2).expect("FATAL: infallible"),
+                BLOCK_MSG_ID,
+            ),
+            self.config.is_mainnet(),
+        );
```

Comment:
This should be a method of `signer_set` like `get_signers_contract(is_mainnet, reward_cycle, msg_id)` -- that way, the miner thread doesn't have to encode logic about the reward cycle modulus.

Reply:
Done